runtime/core/tensor_layout.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/runtime/core/exec_aten/exec_aten.h>
+#include <executorch/runtime/core/exec_aten/util/scalar_type_util.h>
+#include <executorch/runtime/core/span.h>
+#include <executorch/runtime/core/tensor_layout.h>
+
+namespace executorch {
+namespace runtime {
+
+namespace {
+size_t calculate_nbytes(
+    const Span<const int32_t>& sizes,
+    const exec_aten::ScalarType& scalar_type) {
+  ssize_t n = 1;
+  for (ssize_t i = 0; i < sizes.size(); i++) {
+    ET_CHECK(sizes[i] >= 0);
```

Comment:
This will crash if `sizes` contains a negative value, which we need to avoid. See my comment on the .h file about using a static factory, which would then let this helper return `Result<size_t>` to handle the error case non-fatally.

Our core tensor type suffers from the same problem, which is why the runtime has to do an [awkward pre-check](https://github.com/pytorch/executorch/blob/9bb9ad0564453bdbc656ea3502656da9eaa38c28/runtime/executor/tensor_parser_portable.cpp#L108) before creating tensors.